test/unit/Cover/buyCover.js

```diff
@@ -802,4 +805,464 @@ describe('buyCover', function () {
+      },
+      [{ poolId: '0', coverAmountInAsset: amount }],
+      {
+        value: expectedPremium,
+      },
+    );
+
+    const nftBalanceAfter = await coverNFT.balanceOf(coverReceiver.address);
+    expect(nftBalanceAfter).to.be.equal(1);
+
+    const ownerOfCoverId = await coverNFT.ownerOf(coverId);
+    expect(ownerOfCoverId).to.be.equal(coverReceiver.address);
+  });
+
+  it('allows to set a non member as owner', async function () {
+    const { cover, coverNFT } = this;
+
+    const {
+      members: [coverBuyer],
+      nonMembers: [nonMemberCoverReceiver],
+    } = this.accounts;
+
+    const { amount, productId, coverAsset, period, coverId, expectedPremium } = buyCoverFixture;
+
+    const nftBalanceBefore = await coverNFT.balanceOf(nonMemberCoverReceiver.address);
+    expect(nftBalanceBefore).to.be.equal(0);
+
+    await cover.connect(coverBuyer).buyCover(
+      {
+        owner: nonMemberCoverReceiver.address,
+        productId,
+        coverAsset,
+        amount,
+        period,
+        maxPremiumInAsset: expectedPremium,
+        paymentAsset: coverAsset,
+        payWitNXM: false,
+        commissionRatio: parseEther('0'),
+        commissionDestination: AddressZero,
+        ipfsData: '',
+      },
+      [{ poolId: '0', coverAmountInAsset: amount }],
+      {
+        value: expectedPremium,
+      },
+    );
+
+    const nftBalanceAfter = await coverNFT.balanceOf(nonMemberCoverReceiver.address);
+    expect(nftBalanceAfter).to.be.equal(1);
+
+    const ownerOfCoverId = await coverNFT.ownerOf(coverId);
+    expect(ownerOfCoverId).to.be.equal(nonMemberCoverReceiver.address);
+  });
+
+  it('mints rewards to staking pool', async function () {
+    const { cover, tokenController } = this;
+
+    const {
+      governanceContracts: [gv1],
+      members: [coverBuyer, coverReceiver],
+    } = this.accounts;
+
+    const { amount, productId, coverAsset, period, poolId, expectedPremium } = buyCoverFixture;
+
+    const globalRewardsRatio = 5000;
+
+    await cover.connect(gv1).updateUintParameters([1], [globalRewardsRatio]);
+
+    const stakingPoolRewardBefore = await tokenController.stakingPoolNXMBalances(poolId);
+
+    await cover.connect(coverBuyer).buyCover(
+      {
+        owner: coverReceiver.address,
+        productId,
+        coverAsset,
+        amount,
+        period,
+        maxPremiumInAsset: expectedPremium,
+        paymentAsset: coverAsset,
+        payWitNXM: false,
+        commissionRatio: parseEther('0'),
+        commissionDestination: AddressZero,
+        ipfsData: '',
+      },
+      [{ poolId, coverAmountInAsset: amount }],
+      {
+        value: expectedPremium,
+      },
+    );
+
+    const stakingPoolRewardAfter = await tokenController.stakingPoolNXMBalances(poolId);
+    // validate that rewards increased
+    expect(stakingPoolRewardAfter.rewards).to.be.gt(stakingPoolRewardBefore.rewards);
+  });
+
+  it('allows to buy against multiple staking pool', async function () {
```

Comment:
for brevity, let's reduce the tested pools here from 5 to 3

There's no tangible difference in the logic that runs with 5 vs 3

and it makes the test easier to read